buffer/src/main/java/io/netty/buffer/SizeClasses.java

```diff
@@ -237,6 +237,12 @@ private void idx2SizeTab(int[] sizeIdx2sizeTab, int[] pageIdx2sizeTab) {
             int nDelta = sizeClass[NDELTA_IDX];
 
             int size = (1 << log2Group) + (nDelta << log2Delta);
+            if (directMemoryCacheAlignment > 0) {
+                // We need to ensure we align the size before storing it as otherwise we will use the incorrect element
+                // size when creating the PoolSubPage
+                size = alignSize(size);
+            }
```

Comment:
The `PoolArena.HeapArena` is also getting the configured `directMemoryCacheAlignment` for some reason, which I don't think makes sense because then it will also modify the size tables for heap buffers.

Reply:
Fixed it by always using 0 when heap is used.